### PR TITLE
HostToContainer propagation for /sys/fs/bpf

### PR DESCRIPTION
**Pull request by jraby**
By default, mount events from the host namespace are not propagated to a container  after a volume has been mounted into it.
Thus, the initContainer of the agent can never notice changes to /sys/fs/bpf (it cannot detect if the path became a mountpoint)

With mountPropagation: HostToContainer, the mount events are propagated to the container,
allowing the init container to correctly detect changes on /sys/fs/bpf, if it is mounted on the host while the script is blocked waiting for it to be mounted.

In other words, this makes wait-bpf-mount work as advertised.

Signed-off by: Jean Raby <jean@raby.sh>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9575)

### CODE REVIEW

**Review by tgraf — APPROVED** [review]
Great fix

**Comment by tgraf** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

File: install/kubernetes/cilium/charts/agent/templates/daemonset.yaml [hunk 1]
```diff
@@ -256,6 +256,8 @@ spec:
 {{- if not (eq .Values.global.containerRuntime.integration "crio") }}
         - mountPath: /sys/fs/bpf
           name: bpf-maps
+{{- /* Required for wait-bpf-mount to work */ -}}
```
Comment (aanm) [inline install/kubernetes/cilium/charts/agent/templates/daemonset.yaml]: please replace the `*/ -}}` with `*/}}` and regenerate the `quick-install.yaml` file. `cd install/kubernetes && make clean all`

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by jraby** [comment]
Should be in better shape now, I wrote this in github without testing anything :)

Not sure why `enable-debug` was toggled to false or where `      priorityClassName: system-node-critical` comes from, maybe the file had not been regenerated after previous changes?

**Comment by aanm** [comment]
> Should be in better shape now, I wrote this in github without testing anything :)
> 
> Not sure why `enable-debug` was toggled to false or where ` priorityClassName: system-node-critical` comes from, maybe the file had not been regenerated after previous changes?

Yes, they haven't. The `priorityClassName: system-node-critical` needs to be removed manually for now. Sorry

**Comment by aanm** [comment]
test-me-please

**Comment by joestringer** [comment]
So `priorityClassName: system-node-critical` still needs to be removed from the PR? Also would be good to squash the commits into one.

**Comment by jraby** [comment]
Should be in better shape now.

**Review by joestringer — APPROVED** [review]

**Comment by joestringer** [comment]
test-me-please

**Review by aanm — APPROVED** [review]
